src/main/java/org/folio/rest/impl/LoansAPI.java

```diff
@@ -180,6 +186,57 @@ public void postLoanStorageLoansAnonymizeByUserId(
     });
   }
 
+  @Override
+  public void postLoanStorageAnonymizeLoans(List<String> loanIds,
+                                            Map<String, String> okapiHeaders,
+                                            Handler<AsyncResult<Response>> responseHandler,
+                                            Context vertxContext) {
+
+    AnonymizeLoansResponse response = new AnonymizeLoansResponse();
+
+    final Set<String> invalidUuids = loanIds.stream()
+      .filter(id -> !UUIDValidation.isValidUUID(id))
+      .collect(Collectors.toSet());
+
+    if (!invalidUuids.isEmpty()) {
+      final Errors errors = ValidationHelper.createValidationErrorMessage(
+        "invalidLoanIds",
+        Json.encode(invalidUuids),
+        "Loan IDs should be a valid UUID");
+      responseHandler.handle(succeededFuture(
+        PostLoanStorageLoansAnonymizeByUserIdResponse
+          .respond422WithApplicationJson(errors)));
+      return;
+    }
+
+    final String tenantId = TenantTool.tenantId(okapiHeaders);
+    final PostgresClient postgresClient = PgUtil.postgresClient(vertxContext,
+      okapiHeaders);
+
+    final String combinedAnonymizationSql = createAnonymizationSQL(loanIds,
+      tenantId);
+    log.info(String.format("Anonymization SQL: %s", combinedAnonymizationSql));
+
+    executeSql(postgresClient, combinedAnonymizationSql)
+      .map(updateResult -> PostLoanStorageAnonymizeLoansResponse.
+        respond200WithApplicationJson(response.withUpdatedCount(
+          updateResult.getUpdated())))
+      .map(Response.class::cast)
+      .otherwise(this::mapExceptionToResponse)
+      .setHandler(responseHandler);
+
+  }
+
+  private Response mapExceptionToResponse(Throwable t) {
```

Comment:
Seems like this method is redundant. Just use return statement in otherwise section.

Reply:
Done.